crates/wasmparser/src/validator/core.rs

```diff
@@ -817,7 +813,7 @@ impl Module {
             | HeapType::I31 => (),
             HeapType::TypedFunc(type_index) => {
                 // Just check that the index is valid
-                self.func_type_at(type_index.into(), types, offset)?;
+                self.type_at(type_index, offset)?;
```

Comment:
Reading over this again I'm not sure that this is correct? This is verifying that the index is valid but shouldn't the index be further verified? 

I'm not sure if this falls out from type-checking later on or whether it's required happen here. Given that `TypedFunc` should be spelled `Indexed` there's technically not enough context here to verify what the type index should be, so maybe this is ok? I'm not sure if other type-checking contexts are prepared for this validation to no longer happen though.

Reply:
In any case function-references and gc are both behind off-by-default flags, so this can be figured out later if you're unable to bottom it out now.

Reply:
Filed #1039.